lib/puppet/provider/x509_cert/openssl.rb

```diff
@@ -22,11 +22,39 @@ def self.check_private_key(resource)
     cert.check_private_key(priv)
   end
 
+  def self.old_cert_is_equal(resource)
+    cert = OpenSSL::X509::Certificate.new(File.read(resource[:path]))
+
+    altName = ''
+    cert.extensions.each do |ext|
+        altName = ext.value if ext.oid == 'subjectAltName'
+    end
+
+    subjectName = ''
+    cert.subject.to_s.split('/').each do |name|
+        k,v = name.split('=')
+          subjectName = v if k == 'CN'
+    end
+
+    require File.expand_path('../../../../../../inifile/lib/puppet/util/ini_file', __FILE__)
```

Comment:
Need to add ini_file as a requirement for this module for this to work. Also you shouldn't use the full path here because I can specify multiple directories for the module path to fulfill the ini_file requirement and this would fail. I think you can get away with just require 'puppet/util/ini_file' if it is available